### DISCUSSION TITLE
Bug: Scroll resets on key-press when editing system prompt on model creation screen

### DISCUSSION THREAD

**Discussion by willdady** [open-webui/open-webui#7289] [Issues]
## Installation Method

Docker compose

## Environment

- **Open WebUI Version:** v0.4.3
- **Operating System:** MacOS Sequoia
- **Browser (if applicable):** Firefox

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Description

When editing the system prompt on the custom model create/edit screen the scroll constantly resets making it near-impossible to make edits. This is most noticeable when you have a long, multi-paragraph prompt.

**Comment by willdady** [comment]
@tjbck why was this moved from an issue to a discussion? What makes this not a bug?

**Reply by tjbck** [reply]
Have you read the issues template?

```
- **Bug reproducibility**: If a bug cannot be reproduced with a `:main` or `:dev` Docker setup, or a pip install with Python 3.11, it may require additional help from the community. In such cases, we will move it to the "issues" Discussions section due to our limited resources. We encourage the community to assist with these issues. Remember, it’s not that the issue doesn’t exist; we need your help!
```

Hope this helps, would appreciate some help here! :) (I'm also taking a look)

**Reply by willdady** [reply]
Got it! Thanks for the quick reply. Yes I did read the template, but admittedly I made the assumption it was reproducible? I'm running the latest version via docker. I can replicate the issue in both Firefox and Chrome.

**Reply by tjbck** [reply]
More details definitely would've been helpful here, I think I'm able to reproduce the issue you're describing but I'm not 100% sure 😅

**Reply by tjbck** [reply]
Potential fix has been pushed to dev, confirmation wanted here!

**Reply by willdady** [reply]
I still get the same issue when running `docker run -p 3001:8080 -v open-webui:/app/backend/data --name open-webui-dev --add-host=host.docker.internal:host-gateway --restart always ghcr.io/open-webui/open-webui:dev`? 

Trying to run `docker build .` from `dev` yields `ERROR [build 6/6] RUN npm run build`. Not sure what I'm missing 😕

**Reply by silentoplayz** [reply]
Please describe the bug the best you can, even if it necessitates the usage of a large language model to help you clearly articulate the problem.

**Reply by willdady** [reply]
@silentoplayz 

1. Paste a large amount of text (10 paragraphs would be enough) into the system prompt.

2. Scroll down a bit.

3. Try editing the text. Pressing a single key will cause the window scroll to change by itself.

**My guess is the text area is being unmounted/remounted causing a layout reflow. That would explain why the scroll changes.**

**Reply by silentoplayz** [reply]
Based on the provided reproduction steps, I am able to reproduce the issue on the latest dev commit. Safe to say this may be enough to have it on our radar.

**Reply by tjbck** [reply]
I'm 99% positive the issue has been resolved in `:dev` branch.

@willdady Could you share a screen recording of the issue you're describing so we know exactly we're on the same page?

**Reply by willdady** [reply]
@tjbck This recording is from `v0.4.4`. Note every time I press the "a" key the scroll snaps upwards on key-press.

https://github.com/user-attachments/assets/5374f9d2-5c74-4561-ae5c-fe3ee71cc166

**Reply by tjbck** [reply]
Expected behaviour here. That's the current latest dev behaviour as well.

**Comment by willdady** [comment]
@tjbck Just confirming this bug still exists in v0.4.5 [despite the release notes saying it's fixed](https://github.com/open-webui/open-webui/releases/tag/v0.4.5#:~:text=Textarea%20Auto%20Height%20Bug).

**Reply by tjbck** [reply]
Resolved a different issue here 😅, before you weren't even able to edit the contents, whereas now you can at least modify the contents.

**Comment by tjbck** [comment]
@willdady should be entirely fixed on dev, testing wanted here!

**Reply by willdady** [reply]
Scroll issue seems to be fixed on the latest `:dev` image tag. Thank you.
